rust-src/concordium_base/src/hashes.rs

```diff
@@ -23,6 +23,18 @@ pub struct HashBytes<Purpose> {
     _phantom:         PhantomData<Purpose>,
 }
 
+impl<Purpose> PartialEq for HashBytes<Purpose> {
+    fn eq(&self, other: &Self) -> bool {
+        self.bytes == other.bytes && self._phantom == other._phantom
```

Comment:
```suggestion
        self.bytes == other.bytes
```
apart from this it is fine.